src/odemis/acq/stitching/_registrar.py

```diff
@@ -580,8 +581,19 @@ def _insert_tile_to_grid(self, tile):
                     self.tiles[i].append(None)
                     self.shifts_hor[i].append(None)
                     self.shifts_ver[i].append(None)
-        else:
+        elif abs(hor_diff) == 0 and abs(ver_diff) == 0:
             raise ValueError("Cannot insert multiple tiles at the same position.")
+        elif abs(ver_diff) == abs(hor_diff):
+            # Purely diagonal shift. In practice, this will never happen because the smallest difference will be
+            # picked up by one of the previous if-statements. This exception is likely to be triggered only in
+            # artificial tests.
+            # TODO: Implement registrar for tiles which are purely diagonally shifted.
+            raise NotImplementedError("Cannot insert tiles which have a pure diagonal shift, this functionality is "
+                                      "not yet implemented. This error statement is only expected in test cases "
+                                      "since the horizontal shift needs to be exactly equal to the vertical shift.")
+        else:
+            # This should never happen
+            raise ValueError("Unexpected error.")
```

Comment:
For now this code cannot be reached. In case the code above changes, this could be still handy to have, but make the error more descriptive. Add the ver_diff and hor_diff values in the message.

Reply:
I fixed the changes and force pushed on the previous commit. Because it is only a couple of lines I don't think this would make the review process a lot harder plus this way it can be merged directly without merging the commits before finishing the PR

Reply:
Note: When a pr in github is merged, there is a selection, where commits are automatically squashed before the pr is merged. See "squash and merge".